codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketServerProtocolHandler.java

```diff
@@ -57,10 +58,32 @@
     public enum ServerHandshakeStateEvent {
         /**
          * The Handshake was complete succesful and so the channel was upgraded to websockets
+         *
+         * @deprecated in favor of {@link HandshakeComplete} class,
+         * it provides extra information about the handshake
          */
+        @Deprecated
         HANDSHAKE_COMPLETE
```

Comment:
It won't be enough, enum constructors are just for initializing the enum singleton instances. We need to create events dynamically for the selected subprotocols that we process at runtime.